driver-core/src/test/functional/com/mongodb/ClusterFixture.java

```diff
@@ -90,6 +92,8 @@
     private static ConnectionString connectionString;
     private static Cluster cluster;
     private static Cluster asyncCluster;
+    private static Map<ReadPreference, ReadWriteBinding> bindingMap = new HashMap<ReadPreference, ReadWriteBinding>();
```

Comment:
bindings have to be cached now because none of the tests ever release the binding, which would cause an explosion of sessions